code/modules/clothing/head/helmet.dm

```diff
@@ -234,8 +234,8 @@
 
 	return
 
-/obj/item/clothing/head/helmet/ui_action_click(mob/user, actiontype)
-	if(actiontype == /datum/action/item_action/toggle_helmet_flashlight)
+/obj/item/clothing/head/helmet/ui_action_click(mob/user, action)

```

Comment:
yet another reason this system needs to die.
2017 will be the year of real action datums, ```ui_action_click()``` can go suck an egg.